contracts/StakeManager.sol

```diff
@@ -134,6 +154,9 @@ contract StakeManager is Ownable {
      */
     function stake(uint256 _amount, uint256 _timeToIncrease) external onlyVault noPendingMigration finalizeEpoch {
         Account storage account = accounts[msg.sender];
+        if (account.balance > 0) {
+            revert StakeManager__AlreadyStaked();
+        }
```

Comment:
If we're adding this check, then we can also remove the `else{` branch in line `166` that would kick off `processAccount()` because processAccount will then be something that accounts have to call separately, always.

Reply:
Yes, the line 166 is a zombie code that would need to be removed. But more things need to be reviewed to remove restake from same account.